src/D2L.CodeStyle.Analyzers/ApiUsage/Serialization/MustBeDeserializableAnalyzer.cs

```diff
@@ -0,0 +1,574 @@
+					OperationKind.Invocation );
+
+				context.RegisterOperationAction( ( ctx ) => {
+					AnalyzeObjectCreation( ctx, deserializableTypeInfo, additionalFiles );
+				}, OperationKind.ObjectCreation );
+
+				context.RegisterOperationAction( ( ctx ) => {
+					AnalyzeSetters( ctx, deserializableTypeInfo, additionalFiles );
+				}, OperationKind.PropertyReference );
+
+				context.RegisterOperationAction( ( ctx ) => {
+					AnalyzeFieldSetting( ctx, deserializableTypeInfo, additionalFiles );
+				}, OperationKind.FieldReference );
+
+				context.RegisterSymbolAction( ( ctx ) => {
+					AnalyzeMethodDeclaration( ctx, deserializableTypeInfo );
+				}, SymbolKind.Method );
+			}
+		}
+
+		private void AnalyzeMethodDeclaration(
+			SymbolAnalysisContext context,
+			DeserializableTypeInfo deserializableTypeInfo ) {
+			IMethodSymbol? symbol = context.Symbol as IMethodSymbol;
+			if( symbol == null ) {
+				return;
+			}
+
+			IMethodSymbol? interfaceMethod = symbol.ExplicitInterfaceImplementations.FirstOrDefault()
+			                                 ?? symbol.GetImplementedMethods().FirstOrDefault();
+			if( interfaceMethod == null ) {
+				return;
+			}
+
+			// check generic type arguments
+			for( int i = 0; i < symbol.TypeArguments.Length; i++ ) {
+				if( DeserializableAnalyzerHelper.HasMustBeDeserializableAttribute( symbol.TypeArguments[i], deserializableTypeInfo ) ) {
+					if( !DeserializableAnalyzerHelper.HasMustBeDeserializableAttribute( interfaceMethod.TypeArguments[i], deserializableTypeInfo ) ) {
+						context.ReportDiagnostic( Diagnostic.Create( Diagnostics.MustBeDeserializableAttributeShouldBeInTheInterfaceIfInImplementations, symbol.TypeArguments[i].Locations.FirstOrDefault() ) );
+					}
+				}
+			}
+
+			// check method parameters
+			for( int i = 0; i < symbol.Parameters.Length; i++ ) {
+				if( DeserializableAnalyzerHelper.HasMustBeDeserializableAttribute( symbol.Parameters[i], deserializableTypeInfo ) ) {
+					if( !DeserializableAnalyzerHelper.HasMustBeDeserializableAttribute( interfaceMethod.Parameters[i], deserializableTypeInfo ) ) {
+						context.ReportDiagnostic( Diagnostic.Create( Diagnostics.MustBeDeserializableAttributeShouldBeInTheInterfaceIfInImplementations, symbol.Parameters[i].Locations.FirstOrDefault() ) );
+					}
+				}
+			}
+		}
+
+		private void AnalyzeFieldSetting(
+			OperationAnalysisContext context,
+			DeserializableTypeInfo deserializableTypeInfo,
+			ImmutableArray<AdditionalText> additionalFiles ) {
+			IFieldReferenceOperation? operation = context.Operation as IFieldReferenceOperation;
+			if( operation == null || !( operation.Parent is IAssignmentOperation assignmentOperation ) ) {
+				return;
+			}
+
+			SemanticModel? model = assignmentOperation.SemanticModel;
+			ITypeSymbol? type = assignmentOperation.Value.Type;
+
+
+			ISymbol? definitionSymbol = model?.GetSymbolInfo( assignmentOperation.Value.Syntax ).Symbol;
+			if( definitionSymbol == null ) {
+				return;
+			}

```

Comment:
Not sure I'm fully understanding this correctly, but is this equivalent to just calling `operation.Field` to get the field symbol?